docs/leases-and-lease-lifecycle.md

```diff
@@ -0,0 +1,154 @@
+# Leases and Lease Lifecycle
+
+Leases are created in, and leveraged by, KCL to manage a stream's shards.
+This document should help provide insights into the lease lifecycle.
+
+**Nota bene (N.B.):** while actual shard ids are formatted like `shardId-000000000042`, this document uses `shardId[_-]42` for concision.
+
+## Leases
+
+In KCL, a lease provides a temporal assignment between one Kinesis shard and an assigned worker.
+Leases are persistent for the duration of shard processing (detailed later).
+However, lease assignment is transient -- leases may be "stolen" by other workers in the same KCL application.
```

Comment:
> because the assignment has a consistent logic

Pedantically, yes and no. "Yes", the programmed logic doesn't change, yet "no" in that X leases divvied up between Y workers will not produce consistent assignments due to intentional shuffling by KCL.

Per offline consensus, will reword to:
> However, the worker that is processing a lease may change since leases may be "stolen" by other workers in the same KCL application.